airgun/entities/oscapreport.py

```diff
@@ -0,0 +1,79 @@
+from wait_for import wait_for
+
+from airgun.entities.base import BaseEntity
+from airgun.navigation import NavigateStep, navigator
+from airgun.utils import retry_navigation
+from airgun.views.oscapreport import (
+    RemediateModal,
+    SCAPReportDetailsView,
+    SCAPReportView,
+)
+
+
+class OSCAPReportEntity(BaseEntity):
+    endpoint_path = '/compliance/arf_reports'
+
+    def search(self, search_string):
+        """Search for SCAP Report
+
+        :param search_string: how to find the SCAP Report
+        :return: result of the SCAP Report search
+        """
+        view = self.navigate_to(self, 'All')
+        return view.search(search_string)
+
+    def details(self, search_string, widget_names=None, limit=None):
+        """Read the content from corresponding SCAP Report dashboard,
+            clicking on the link in Reported At column of
+            SCAP Report list
+
+        :param search_string:
+        :param limit: how many rules results to fetch at most
+        :return: list of dictionaries with values from SCAP Report Details View
+        """
+        view = self.navigate_to(self, 'Details', search_string=search_string)
+        return view.read(widget_names=widget_names, limit=limit)
+
+    def remediate(self, search_string, resource):
+        """Remediate the failed rule using automatic remediation through Ansible
+
+        :param search_string:
+        """
+        view = self.navigate_to(self, 'Details', search_string=search_string)
+        view.table.row(resource=resource).actions.fill('Remediation')
+        view = RemediateModal(self.browser)
+        view.wait_displayed()
+        self.browser.plugin.ensure_page_safe()
+        wait_for(lambda: view.title.is_displayed, timeout=10, delay=1)
+        view.fill({'select_remediation_method.snippet': 'Ansible'})
```

Comment:
`Ansible` is a default value? you could expect some more options there ?

Reply:
That's what this function does as per docstring: `Remediate the failed rule using automatic remediation through Ansible`.
It surely can be changed to allow for different methods. On one hand, it's not a goal of this function/test to test various methods of remote execution. On the other hand, there is a wizard specifically for remediation that lets you choose so IMO this will be a good addition, later.